src/dockerhub.js

```diff
@@ -0,0 +1,37 @@
+var rp = require( "request-promise" );
+var urlencode = require( "urlencode" );
+var when = require( "when" );
+
+function onRequest( body ) {
+	return body;
+}
+
+function onError( err ) {
+	console.error( "There was an error checking a tags existance on docker hub, ", err );
+	return [];
+}
+
+function listTags( namesapce, name ) {
+	var options = {
+		uri: "https://registry.hub.docker.com/v1/repositories/" + urlencode( namesapce ) + "/" + urlencode( name ) + "/tags",
```

Comment:
this should be a format string. that will let you define the url at the module scope like this:

`https://registry.hub.docker.com/v1/repositories/%s/%s/tags`

and then format it when the `listTags` function is called like this:

`util.format(uri,urlencode( namesapce ),urlencode( name ))`